backend/src/main.py

```diff
@@ -25,3 +26,60 @@ async def lifespan(app: FastAPI):
 @app.get("/")
 def hello():
     return {"Hello": "World"}
+
+# write FastAPI endpoint to get listing record for ID
+# example: GET /listing?id=1
+# response: {"id": 1, "name": "Listing 1", "description": "Description of listing 1"}
+# if ID does not exist, return 404 status code
+# if ID is not provided, return 400 status code
+
+# Mock database
+listings = {
+    1: {
+        "id": 1,
+        "name": "Listing 1",
+        "description": "Description of listing 1",
+        "beds": 2,
+        "bedrooms": 2,
+        "mean_rating": 4.5,
+        "count_ratings": 100
+    },
+    2: {
+        "id": 2,
+        "name": "Listing 2",
+        "description": "Description of listing 2"
+        "beds": 3,
+        "bedrooms": 3,
+        "mean_rating": 4.0,
+        "count_ratings": 50
+    },
+    # Add more listings as needed
+}
+
+@app.get("/listing")
+def get_listing(id: Optional[int] = Query(None, description="The ID of the listing to retrieve")):
```

Comment:
This should use a Pydantic model to ensure the input data matches correctly so we're not injecting technical logic about whether the ID exists into the endpoint's business logic.

Reply:
Let's also use a response model that should match the type structure of a listing object.